src/shared/components/home/emojis-form.tsx

```diff
@@ -0,0 +1,431 @@
+import { EmojiMart } from "../common/emoji-picker";
+import { HtmlTags } from "../common/html-tags";
+import { Icon } from "../common/icon";
+import { Subscription } from "rxjs";
+import { pictrsUri } from "../../env";
+
+interface EmojiFormState {
+    siteRes: GetSiteResponse;
+    custom_emojis: CustomEmojiViewForm[];
+    loading: boolean;
+}
+
+class CustomEmojiViewForm {
+    id: number;
+    category: string;
+    shortcode: string;
+    image_url: string;
+    alt_text: string;
+    keywords: string;
+    changed: boolean;
+  }
+
+export class EmojiForm extends Component<any, EmojiFormState> {
+    private isoData = setIsoData(this.context);
+    private subscription: Subscription | undefined;
+    private emptyState: EmojiFormState = {
+        loading: false,
+        siteRes: this.isoData.site_res,
+        custom_emojis: this.isoData.site_res.custom_emojis.map(x =>
+        ({
+            id: x.custom_emoji.id,
+            category: x.custom_emoji.category,
+            shortcode: x.custom_emoji.shortcode,
+            image_url: x.custom_emoji.image_url,
+            alt_text: x.custom_emoji.alt_text,
+            keywords: x.keywords.map(x => x.keyword).join(" "),
+            changed: false,
+        }))
+    };
+    private emojiLookup = {};
+    private scrollRef: any = {};
+    constructor(props: any, context: any) {
+        super(props, context);
+        this.state = this.emptyState;
+
+        this.parseMessage = this.parseMessage.bind(this);
+        this.subscription = wsSubscribe(this.parseMessage);
+
+        this.emojiLookup = getCustomEmojiLookup();
+    }
+    get documentTitle(): string {
+        return i18n.t("custom_emojis");
+    }
+
+    componentWillUnmount() {
+        if (isBrowser()) {
+          this.subscription?.unsubscribe();
+        }
+      }
+
+    render() {
+        return (
+            <div className="col-12">
+                <Prompt
+                    when={!this.state!.loading}
+                    message={i18n.t("block_leaving")}
+                />
+                <HtmlTags
+                    title={this.documentTitle}
+                    path={this.context.router.route.match.url}
+                />
+                <h5 className="col-12">{i18n.t("custom_emojis")}</h5>
+                {Object.keys(this.emojiLookup).length > 0 && <div>
+                    <EmojiMart onEmojiClick={this.handleEmojiClick} pickerOptions={this.configurePicker()}></EmojiMart>
+                </div>}
+                <div className="table-responsive col-12">
+                    <table
+                        id="emojis_table"
+                        className="table table-hover"
+                    >
+                        <thead className="pointer">
+                            <tr>
+                                <th style="width:70px;">{i18n.t("column_emoji")}</th>
```

Comment:
Re-do this to bootstrap columns, rather than a table. It'll make it more responsive and usable on mobile, and you won't have to hard-code widths. I need to get rid of every html table in lemmy-ui.

Reply:
I'm gonna have to push back on this one. Tables are the correct way to display this kind of data. Moving to bootstrap columns would end up wasting a ton of space, as 12 columns isn't specific enough about widths. None of the columns need to be responsive, except for the keywords column, which is set up to expand to fill any remaining width.

I have made a change to optimize it a bit better for mobile by giving the table a min-width so that keywords isn't messed up, but its as usable as its ever gonna be for a small device.

Reply:
Okay, but at the very least remove the hard-coded widths ( those aren't going to be correct for any device but the one you coded them from), and use bootstrap table directives like in `communities.tsx` . As I said though, tables are really terrible for skinny devices like phones, which is why we eventually need to get rid of that, and switch to using responsive design.

Bootstrap has responsive form examples, which is what this is, and we have a good amount of responsive forms in lemmy.